pkg/apis/tensorflow/v1alpha1/defaults.go

```diff
@@ -0,0 +1,77 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/tensorflow/k8s/pkg/util"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+
+func addDefaultingFuncs(scheme *runtime.Scheme) error {
+	return RegisterDefaults(scheme)
+}
+
+
+// SetDefaults sets any unspecified values to defaults
+func SetDefaults_TfJob(obj *TfJob) {
+	c := obj.Spec
+
+	if c.TfImage == "" {
+		c.TfImage = DefaultTFImage
+	}
+
+	// Check that each replica has a TensorFlow container.
+	for _, r := range c.ReplicaSpecs {
+
+		if r.TfPort == nil {
+			r.TfPort = proto.Int32(TfPort)
+		}
+
+		if string(r.TfReplicaType) == "" {
+			r.TfReplicaType = MASTER
+		}
+
+		if r.Replicas == nil {
+			r.Replicas = proto.Int32(Replicas)
+		}
+
+		//Set the default configuration for a PS server if the user didn't specify a PodTemplateSpec
+		if r.Template == nil && r.TfReplicaType == PS {
+			setDefault_PSPodTemplateSpec(r, c.TfImage)
+		}
+	}
+	if c.TerminationPolicy == nil {
+		c.TerminationPolicy = &TerminationPolicySpec{
+			Chief: &ChiefSpec{
+				ReplicaName:  "MASTER",
+				ReplicaIndex: 0,
+			},
+		}
+	}
+
+	return nil
+}
+
+func setDefault_PSPodTemplateSpec(r *TfReplicaSpec, tfImage string) {
```

Comment:
seems this func not join default generator, we can move this？

Reply:
@zjj2wry The `setDefault_PSPodTemplateSpec` is called by the `SetDefaults_TfJob` func. I think we should keep it.

Reply:
you are right, i miss this~